pubspec.yaml

```diff
@@ -4,28 +4,28 @@ description: General utilities for Dart projects.
 homepage: https://www.github.com/Workiva/w_common
 
 environment:
-  sdk: '>=2.4.0 <3.0.0'
+  sdk: '>=2.9.0 <3.0.0'
 
 dependencies:
-  args: ^1.5.2
+  args: '>=1.5.2 <3.0.0'
   async: ^2.4.0
-  colorize: ^2.0.0
-  glob: ^1.2.0
+  colorize: ^3.0.0
+  file: ^6.0.0
+  glob: ^2.0.0
   intl: ">=0.15.0 <0.18.0"
   js: ^0.6.1+1
-  logging: ^0.11.3+2
+  logging: '>=0.11.3+2 <3.0.0'
   meta: ^1.1.8
-  package_config: ^1.1.0
-  package_resolver: ^1.0.10
+  package_config: ^2.0.0
   path: ^1.6.4
-  sass: ^1.23.3
+  sass: '>=1.32.11 <2.0.0'
   source_maps: ^0.10.8
-  watcher: ^0.9.7+12
+  watcher: '>=0.9.7+12 <2.0.0'
 
 dev_dependencies:
-  build_runner: ^1.7.1
-  build_test: ^0.10.9
-  build_web_compilers: ^2.5.1
-  dependency_validator: ^1.4.0
+  build_runner: '>=1.7.1 <3.0.0'
+  build_test: '>=0.10.9 <3.0.0'
+  build_web_compilers: '>=2.5.1 <4.0.0'
+  dependency_validator: '>=1.4.0 <4.0.0'
```

Comment:
Will this actually resolve with some of the lower versions of the version ranges here? Wondering if we can just use `^` to limit to a single major version for most/all of these.

Reply:
We can't do that for all of then, e.g. @#$@ sass made a breaking change in a minor version, and the build packages are weird. But I did a pub downgrade and raised the limits to what that will support.